server/src/main/java/org/opensearch/indices/recovery/RecoverySettings.java

```diff
@@ -87,10 +88,10 @@ public class RecoverySettings {
     /**
      * Controls the maximum number of streams that can be started concurrently per recovery when downloading from the remote store.
      */
-    public static final Setting<Integer> INDICES_RECOVERY_MAX_CONCURRENT_REMOTE_STORE_STREAMS_SETTING = Setting.intSetting(
+    public static final Setting<Integer> INDICES_RECOVERY_MAX_CONCURRENT_REMOTE_STORE_STREAMS_SETTING = new Setting<>(
         "indices.recovery.max_concurrent_remote_store_streams",
-        10,
-        1,
+        (s) -> Integer.toString(Math.max(1, OpenSearchExecutors.allocatedProcessors(s) / 2)),
```

Comment:
Thanks @andrross, since we are not CPU bound (for this specific workload), could we base our decision on how much heap not is allocated? For example, one stream needs a headway of 100Mb , let's take 50% of JVM heap and divide by 100Mb. What do you think?